index.test.js

```diff
@@ -56,7 +67,133 @@ describe('instance', () => {
           .matters(Equivalency.en.COMMON_PUNCTUATION);
         const inputs = [['what he did.', 'what he did?']];
         inputs.forEach(([s1, s2]) => {
-          expect(instance.equivalent(s1, s2)).toEqual({ isEquivalent: false });
+          expect(instance.equivalent(s1, s2, { giveReasons: true })).toEqual({
+            isEquivalent: false,
+            reasons: [{ name: 'common punctuation' }],
+          });
+        });
+      });
+
+      it('identifies multiple rules that are reasons (punctuation and symbols)', () => {
+        const instance = new Equivalency()
+          .matters(Equivalency.en.COMMON_PUNCTUATION)
+          .matters(Equivalency.en.COMMON_SYMBOLS)
+          .doesntMatter(Equivalency.WHITESPACE_DIFFERENCES);
+        const correctAnswer = 'you and me';
+
+        expect(
+          instance.equivalent(correctAnswer, 'you and me!', {
+            giveReasons: true,
+          })
+        ).toEqual({
+          isEquivalent: false,
+          reasons: [{ name: 'common punctuation' }],
+        });
+
+        expect(
+          instance.equivalent(correctAnswer, 'you &and me', {
+            giveReasons: true,
+          })
+        ).toEqual({
+          isEquivalent: false,
+          reasons: [{ name: 'common symbols' }],
+        });
+
+        // If these are applied together, passes, else doesn't.
+        expect(
+          instance.equivalent(correctAnswer, 'you &and me!', {
+            giveReasons: true,
+          })
+        ).toEqual({
+          isEquivalent: false,
+          reasons: [{ name: 'common punctuation' }, { name: 'common symbols' }],
+        });
+
+        expect(
+          instance.equivalent(correctAnswer, 'you and I', {
+            giveReasons: true,
+          })
+        ).toEqual({
+          isEquivalent: false,
+          reasons: [{ name: 'identity' }],
+        });
+
+        expect(
+          instance.equivalent(correctAnswer, 'you &and I!', {
+            giveReasons: true,
+          })
+        ).toEqual({
+          isEquivalent: false,
+          reasons: [{ name: 'identity' }],
+        });
+      });
+
+      it('gives empty array of reasons when giveReasons: true and isEquivalent: true', () => {
+        const instance = new Equivalency()
+          .doesntMatter(Equivalency.UNICODE_NORMALIZATION)
+          .doesntMatter(Equivalency.WHITESPACE_DIFFERENCES);
+        const correctAnswer = 'aeiou';
+
+        expect(
+          instance.equivalent(correctAnswer, 'aeiou', {
+            giveReasons: true,
+          })
+        ).toEqual({
+          isEquivalent: true,
+          reasons: [],
+        });
+      });
+
+      it('identifies multiple rules that are reasons (grave accent, umlaut, other diacritic)', () => {
```

Comment:
This is the test that is most directly tied to what we'd use this change for.

Reply:
Really shows how much simpler the approach for showing the right hint will be.